src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheck.java

```diff
@@ -147,6 +180,49 @@
  *   &lt;property name=&quot;allowSamelineParameterizedAnnotation&quot; value=&quot;true&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>
+ * Example to allow only one and only parameterized annotation on the same line:
+ * </p>
+ * <pre>
+ * &#64;NotNull private boolean field1; //violation
+ * &#64;Override public int hashCode() { return 1; } //violation
+ * &#64;NotNull //ok
+ * private boolean field2;
+ * &#64;Override //ok
+ * public boolean equals(Object obj) { return true; }
+ * &#64;Mock DataLoader loader; //violation
+ * &#64;SuppressWarnings("deprecation") DataLoader loader; //ok
+ * &#64;SuppressWarnings("deprecation") public int foo() { return 1; } //ok
+ * &#64;SuppressWarnings("deprecation") &#64;Mock DataLoader loader; //violation
+ * </pre>
+ * <p>
+ * Use the following configuration to only validate annotations on methods to allow one
+ * single parameterless annotation on the same line:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;AnnotationLocation&quot;&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;METHOD_DEF&quot;/&gt;
+ *   &lt;property name=&quot;allowSamelineMultipleAnnotations&quot; value=&quot;false&quot;/&gt;
+ *   &lt;property name=&quot;allowSamelineSingleParameterlessAnnotation&quot;
+ *     value=&quot;true&quot;/&gt;
```

Comment:
please recheck if we can `property` tag on single line. If linelength is ok, please make it for all blocks.

Reply:
Not possible, it will make build fail.